tests/integration/ha_tests/test_ha.py

```diff
@@ -317,3 +377,57 @@ async def test_restart_db_process(ops_test, continuous_writes, change_logging):
 
     # verify that old primary is up to date.
     await verify_writes(ops_test)
+
+
+async def test_network_cut(ops_test: OpsTest, continuous_writes, chaos_mesh):
+    app = await get_application_name(ops_test, APP_NAME)
+    primary = await get_replica_set_primary(ops_test)
+
+    primary_unit = None
+    active_unit = None
+    for unit in ops_test.model.applications[app].units:
+        if unit.name == primary:
+            primary_unit = unit
+        else:
+            active_unit = unit
+        if primary_unit and active_unit:
+            break
```

Comment:
I think it would be nice if this function could return the unit itself. That way you could do:
```
# retrieve a primary unit and a non-primary unit (active-unit). The primary unit will have its network disrupted, while the active unit allows us to communicate to `mongod`
primary = await get_replica_set_primary(ops_test)
active_unit = list(set(ops_test.model.applications[app].units) - set([primary]))[0]
```